cuda/btx_cudamatching_model.yaml

```diff
@@ -0,0 +1,293 @@
+      - :name: ^pctx_val$
+  
+  # Device primary context management
+  - :set_id: primary_ctx_retain
+    :name: "cuDevicePrimaryCtxRetain"
+    :register: false
+  - :set_id: primary_ctx_retain_entry
+    :domain: entries & primary_ctx_retain
+    :payload_field_class:
+      :type: structure
+      :members:
+      - :name: ^dev$
+  - :set_id: primary_ctx_retain_exit
+    :domain: exits & primary_ctx_retain
+    :payload_field_class:
+      :type: structure
+      :members:
+      - :name: ^cuResult$
+      - :name: ^pctx_val$
+  
+  - :set_id: primary_ctx_release
+    :name: "cuDevicePrimaryCtxRelease"
+    :register: false
+  - :set_id: primary_ctx_release_entry
+    :domain: entries & primary_ctx_release
+    :payload_field_class:
+      :type: structure
+      :members:
+      - :name: ^dev$
+  - :set_id: primary_ctx_release_exit
+    :domain: exits & primary_ctx_release
+    :payload_field_class:
+      :type: structure
+      :members:
+      - :name: ^cuResult$
+  
+  - :set_id: primary_ctx_reset
+    :name: "cuDevicePrimaryCtxReset"
+    :register: false
+  - :set_id: primary_ctx_reset_entry
+    :domain: entries & primary_ctx_reset
+    :payload_field_class:
+      :type: structure
+      :members:
+      - :name: ^dev$
+  - :set_id: primary_ctx_reset_exit
+    :domain: exits & primary_ctx_reset
+    :payload_field_class:
+      :type: structure
+      :members:
+      - :name: ^cuResult$
+
+  # Stream create
+  - :set_id: stream_create
+    :name: "cuStreamCreate"
+    :register: false
+  - :set_id: stream_create_entry
+    :domain: entries & stream_create
+    :register: false
+  - :set_id: stream_create_exit
+    :domain: exits & stream_create
+    :payload_field_class:
+      :type: structure
+      :members:
+      - :name: ^cuResult$
+      - :name: ^phStream_val$
+
+  # kernel function name tracking
+  - :set_id: module_get_function
+    :name: "cuModuleGetFunction"
+    :register: false
+  - :set_id: module_get_function_entry
+    :domain: entries & module_get_function
+    :payload_field_class:
+      :type: structure
+      :members:
+      - :name: ^name_val$
+  - :set_id: module_get_function_exit
+    :domain: exits & module_get_function
+    :payload_field_class:
+      :type: structure
+      :members:
+      - :name: ^cuResult$
+      - :name: ^hfunc_val$
+
+  # tasks that generate thapi device events
+  - :set_id: task
+    :name: "(cuMemcpy|cuMemPrefetch|cuMemset|cuMemFree|cuLaunch)"
+    :register: false
+
+  - :set_id: task_entry
+    :domain: entries & task
+    :register: false
+
+  - :set_id: task_stream_present_entry
+    :domain: task_entry
+    :register: false
+    :payload_field_class:
+      :type: structure
+      :members:
+      - :name: ^hStream$
+
+  - :set_id: task_stream_absent_entry
+    :domain: task_entry - task_stream_present_entry
+    :register: false
+
+  # specialize for whether a funtion arg is present or not
+  - :set_id: kernel_task_entry
+    :name: "cuLaunch[^H]"
```

Comment:
Maybe a comment tp explain why not the `H`? For Host i  guess :)

Reply:
added a comment, agreed it is not obvious